sources/data-node/api/v2/trading_data.proto

```diff
@@ -107,6 +111,39 @@ message PageInfo {
     string end_cursor = 4;
 }
 
+// -- Positions --
+// Request for a list of positions for a party
+// Optionally, if a market identifier is set, the results will be filtered for that market only
+message GetPositionsByPartyRequest {
+    // Party identifier, required field
+    string party_id = 1;
+    // Market identifier
+    string market_id = 2;
+}
+
+// Response for a list of positions for a party
+message GetPositionsByPartyResponse {
+    // A list of 0 or more positions
+    repeated vega.Position positions = 1;
+}
+
+// Request for a list of positions for a party
+// Optionally, if a market identifier is set, the results will be filtered for that market only
+message GetPositionsByPartyPagedRequest {
+    // Party identifier, required field
+    string party_id = 1;
+    // Market identifier
+    string market_id = 2;
+    // Optional pagination control
+    optional Pagination pagination = 3;
+}
+
+// Response for a list of positions for a party
+message GetPositionsByPartyPagedResponse {
+    // A list of 0 or more positions
+    PositionConnection positions = 1;
+}
+
```

Comment:
In accordance with the suggested guidelines defined by Emil, we should put these definitions in the order of the rpc function definition, e.g. if you define the rpc function immediately below orders, then these definitions should be declared immediately after the order message definitions.